collector-api/internal/api/snapshots.go

```diff
@@ -195,5 +207,74 @@ func handleCompactSnapshot(s3Location string, collectedAt int64) error {
 	// Here, implement your actual logic for processing the compact snapshot
 	// e.g., saving to local storage, uploading to S3, etc.
 	log.Printf("Processing compact snapshot with s3_location: %s and collected_at: %d", s3Location, collectedAt)
+
+	f, err := os.Open(path.Join("/usr/local/autodba/share/collector_api_server/storage", s3Location))
```

Comment:
We cannot assume the downloaded file is always here. I'm thinking maybe we should change this line:

https://github.com/crystaldb/autodba/blob/292d1b346f15627e1bcfb11a16976d826fb4c25e/collector-api/internal/api/upload.go#L93

with 

```
	resp := s3UploadResponse{Key: filePath}
```

This way, you'll receive the absolute path in `s3Location` variable. Can you please apply this change and check it?